src/shell.zig

```diff
@@ -73,7 +73,7 @@ pub fn echo(shell: *Shell, comptime fmt: []const u8, fmt_args: anytype) void {
         }
         pos += 1;
     }
-    fmt_ansi = fmt_ansi ++ fmt[pos_start..pos] ++ "\n";
+    fmt_ansi = fmt_ansi ++ fmt[pos_start..@min(pos, fmt.len)] ++ "\n";
```

Comment:
Wait what? This is compiler bug, right? It _shouldn't_ be incrementing `pos += 1` on last time? Maybe

```zig
    next_pos: inline while (pos < fmt.len) {
        if (fmt[pos] == '{') {
            inline for (comptime std.meta.fieldNames(@TypeOf(ansi))) |field_name| {
                const tag = comptime "{ansi-" ++ field_name ++ "}";
                if (comptime std.mem.startsWith(u8, fmt[pos..], tag)) {
                    fmt_ansi = fmt_ansi ++ fmt[pos_start..pos] ++ @field(ansi, field_name);
                    pos += tag.len;
                    pos_start = pos;
                    continue :next_pos;
                }
            } else {
                pos += 1;
            }
        } else {
            pos += 1;
        }
    }
    // Note: We duplicate `pos += 1` above, otherwise this hits a compiler bug. 
    comptime assert(pos == fmt.len);
    fmt_ansi = fmt_ansi ++ fmt[pos_start..pos] ++ "\n";
```

instead? 

Sorry for writing convoluted code which confused the compiler :fearful:

Reply:
I think continues/breaks inside `inline for/while` are not handled properly